### PR TITLE
added URL filters

### LINKED ISSUE #2563: Assess QASP owasp scan accuracy
## Description
QASP scan is currently scanning 455 URLs, mostly unrelated to our app. Please ensure we only scan relevant URLs.

## Action Taken
n/a

## What I expected to see
Only TDP URLs. Scan is currently hitting 455 URLs including youtube etc.

## What I did see
https://github.com/raft-tech/TANF-app/pull/2542#pullrequestreview-1466036195

## Other Helpful Information
It looks like knowledge center updates are adding to the URLs here (crazyegg, youtube, JS libs). Perhaps if we can ignore the knowledge center it will revert to previous number of URLs.
 
- Backend scan result 
[ZAP Scanning Report _ Backend.pdf](https://github.com/raft-tech/TANF-app/files/11708305/ZAP.Scanning.Report._.Backend.pdf)

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2563 _

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: scripts/zap-scanner.sh [hunk 1]
```diff
@@ -59,15 +59,91 @@ ZAP_CLI_OPTIONS="\
...
   -config globalexcludeurl.url_list.url\(2\).regex='^https?://.*\.amazonaws\.(?:com|org|net)/.*$' \
   -config globalexcludeurl.url_list.url\(2\).description='TDP S3 buckets' \
   -config globalexcludeurl.url_list.url\(2\).enabled=true \
-  -config globalexcludeurl.url_list.url\(3\).regex='^https:\/\/.*\.acf.hhs.gov\/v1\/login\/.*$' \
-  -config globalexcludeurl.url_list.url\(3\).description='Site - identity pages' \
+
+  -config globalexcludeurl.url_list.url\(3\).regex='^https:\/\/.*\.hhs.gov\/.*$' \
+  -config globalexcludeurl.url_list.url\(3\).description='Site - acf.hhs.gov' \
   -config globalexcludeurl.url_list.url\(3\).enabled=true \
+
```
Comment (raftmsohani) [inline scripts/zap-scanner.sh:74]: Investigated an approach to get rid of these urls ending up in the scan, unfortunately couldn't find any way to skip these

**Review by andrew-jameson — APPROVED** [review]

**Review by George-Hudson — APPROVED** [review]

**Comment by ADPennington** [comment]
is this PR in qasp review @raftmsohani @Smithh-Co ?

**Comment by Smithh-Co** [comment]
@ADPennington Hi Alex, I believe so - pulled it into the QASP column but I can't seem to add you as a reviewer.

@raftmsohani FYI Mo

**Comment by ADPennington** [comment]
is the `qasp review` [label](https://github.com/raft-tech/TANF-app/blob/develop/.github/workflows/qasp-owasp-scan.yml) not triggering the owasp scan anymore @George-Hudson?

**Comment by George-Hudson** [comment]
> is the `qasp review` [label](https://github.com/raft-tech/TANF-app/blob/develop/.github/workflows/qasp-owasp-scan.yml) not triggering the owasp scan anymore @George-Hudson?

@ADPennington it had not... when we changed over we wanted github to tell circle ci it was triggering a run, but we had not done that for QASP (nor for labeled deploys). I created a PR into this branch to fix this issue: [here](https://github.com/raft-tech/TANF-app/pull/2601), which, you can see allows the OWASP scan to run in [circleci](https://app.circleci.com/pipelines/github/raft-tech/TANF-app/14181)

Good catch.

**Review by ADPennington — APPROVED** [review]
great to see only a small list of URLs now! Thanks @raftmsohani @George-Hudson!

2558p1

**Pull request merged**